storage/pipeline/input.go

```diff
@@ -792,8 +792,16 @@ func (m *Sealing) tryGetDealSector(ctx context.Context, sp abi.RegisteredSealPro
 		return xerrors.Errorf("getting storage config: %w", err)
 	}
 
-	// if we're above WaitDeals limit, we don't want to add more staging sectors
-	if cfg.MaxWaitDealsSectors > 0 && m.stats.curStaging() >= cfg.MaxWaitDealsSectors {
+	fullSizePieceIsPending := false
+	for _, piece := range m.pendingPieces {
+		// Check if any pending pieces are full size
+		if piece.size == abi.PaddedPieceSize(ssize).Unpadded() {
```

Comment:
I'm a little fuzzy on the math here. If we have a `32GiB` sector pending and the sector size is `32GiB`, will be padding work out and this will be true?

Reply:
Yep, this is correct